core/src/test/java/software/amazon/awssdk/core/auth/EC2MetadataServiceMock.java

```diff
@@ -171,6 +173,7 @@ public void run() {
                 } finally {
                     try {
                         outputStream.close();
+                        socket.close();
```

Comment:
If the first close throws this won't get cleaned up

Reply:
good point, fixed!